tests/mpi/interpolate_to_different_mesh_01.cc

```diff
@@ -0,0 +1,197 @@
+	filenames.push_back (filename_ +
+			     Utilities::int_to_string (loop, 2) +
+			     "." +
+			     Utilities::int_to_string(i, 2) +
+			     ".vtu");
+      const std::string
+	pvtu_master_filename = (filename_ +
+				Utilities::int_to_string (loop, 2) +
+				".pvtu");
+      std::ofstream pvtu_master (pvtu_master_filename.c_str());
+      data_out.write_pvtu_record (pvtu_master, filenames);
+    }
+}
+
+template<int dim>
+void test()
+{
+  unsigned int myid = Utilities::MPI::this_mpi_process (MPI_COMM_WORLD);
+
+  parallel::distributed::Triangulation<dim> tr(MPI_COMM_WORLD,
+					       dealii::Triangulation<dim,dim>::none,
+					       parallel::distributed::Triangulation<dim>::no_automatic_repartitioning);
+
+  GridGenerator::hyper_cube(tr);
+  tr.refine_global(2);
+  parallel::distributed::Triangulation<dim> tr2(MPI_COMM_WORLD,
+					       dealii::Triangulation<dim,dim>::none,
+					       parallel::distributed::Triangulation<dim>::no_automatic_repartitioning);
+
+  GridGenerator::hyper_cube(tr2);
+  tr2.refine_global(2);
+  
+  FE_Q<dim> fe(1);
+  DoFHandler<dim> dh(tr);
+  DoFHandler<dim> dh2(tr2);
+
+  SomeFunction<dim> func;
+  
+
+  for (unsigned int loop = 0; loop < 5; ++loop)
+    {
+      // randomly refine:
+      std::vector<bool> r_flags(tr.n_active_cells()*dim, false);
+      std::vector<bool> c_flags(tr.n_active_cells(), false);
+
+      for (unsigned int i=0;i<c_flags.size();++i)
+	{
+	  int roll = Testing::rand()%4;
+	  if (roll >= 2)
+	    {
+	      for (unsigned int j=0;j<dim;++j)
+		r_flags[i*dim+j] = true;
+	    }
+	  else if (roll == 1)
+	    c_flags[i] = true;
+	}
+
+      tr.load_coarsen_flags(c_flags);
+      tr.load_refine_flags(r_flags);
+
+      tr.execute_coarsening_and_refinement ();
+      deallog << "locally owned cells: " << tr.n_locally_owned_active_cells()
+	      << " / "
+	      << tr.n_global_active_cells()
+	      << std::endl;
+
+
+      LA::MPI::Vector vec1;
+      LA::MPI::Vector lr_vec1;
+      setup(dh, fe, vec1, lr_vec1);
+
+      LA::MPI::Vector vec2;
+      LA::MPI::Vector lr_vec2;
+      setup(dh2, fe, vec2, lr_vec2);
+
+      // interpolate func on old mesh:
+      VectorTools::interpolate(dh2, func, vec2);
+      lr_vec2 = vec2;
+
+      // interpolate from vec2 to vec1
+      VectorTools::interpolate_to_different_mesh(dh2, lr_vec2, dh, vec1);
+      lr_vec1 = vec1;
+
+      {
+	Vector<double> local_errors (tr.n_active_cells());
+	VectorTools::integrate_difference	(dh, lr_vec1, func, local_errors,
+						 QGauss<dim>(3), VectorTools::L2_norm);
+	double total_local_error = local_errors.l2_norm();
+	const double total_global_error
+	  = std::sqrt (Utilities::MPI::sum (total_local_error * total_local_error, MPI_COMM_WORLD));
+	if (myid == 0)
+	  deallog << "err: " << total_global_error << std::endl;
+      }
+      
+      //output(dh, lr_vec1, loop, "solutionA-");
+      //output(dh2, lr_vec2, loop, "solutionB-");
+            
```

Comment:
everything after this point no longer contributes to the mission of the testcase, or do I miss something?